exe-common/src/sync.rs

```diff
@@ -404,13 +406,17 @@ fn perform_rollback(
         // Either we are syncing historical data and rollback is in old epoch
         // Or we dropped all loose blocks and now latest tip is in packed epoch
         let current_epoch = last_date.get_epochid();
+        assert_ne!(current_epoch, 0, "Cannot roll back first epoch");
         match storage.write() {
             Ok(mut storage) => {
                 let last_packed_epoch = (storage.packed_epochs_len() - 1) as u64;
                 if current_epoch == last_packed_epoch {
                     // We are inside last packed epoch
                     // Drop current epoch and roll back to previous one
-                    storage.drop_packed_epoch(current_epoch)?;
+                    // Note: last_block must be the last block in an epoch in order for it to find the correct
+                    //       chain_state hash to delete.
+                    assert_eq!(last_date.slotid(), Some((net_cfg.epoch_stability_depth * 10 - 1) as u16));
```

Comment:
Another option here would be to remove this check, and then make removal of old chainstate data on disk (should be inaccessible) best-effort in case last_block for some reason is not the last of an epoch (since this is what is hashed to get the identifier for chainstate local storage).